lightwood/api/json_ai.py

```diff
@@ -369,96 +404,94 @@ def add_implicit_values(json_ai: JsonAI) -> JsonAI:
-                    "data": "data",
-                    "dtype_dict": "$dtype_dict",
-                    "target": "$target",
-                },
-            }
-
+    }), ('analyzer', {
+         "module": "model_analyzer",
+         "args": {
+             "stats_info": "$statistical_analysis",
+             "ts_cfg": "$problem_definition.timeseries_settings",
+             "accuracy_functions": "$accuracy_functions",
+             "predictor": "$ensemble",
+             "data": "test_data",
+             "train_data": "train_data",
+             "target": "$target",
+             "disable_column_importance": "False",
+             "dtype_dict": "$dtype_dict",
+             "fixed_significance": None,
+             "confidence_normalizer": False,
+             "positive_domain": "$statistical_analysis.positive_domain",
+         },
+         }), ('explainer', {
+             "module": "explain",
+             "args": {
+                 "timeseries_settings": "$problem_definition.timeseries_settings",
+                 "positive_domain": "$statistical_analysis.positive_domain",
+                 "fixed_confidence": "$problem_definition.fixed_confidence",
+                 "anomaly_detection": "$problem_definition.anomaly_detection",
+                 "anomaly_error_rate": "$problem_definition.anomaly_error_rate",
+                 "anomaly_cooldown": "$problem_definition.anomaly_cooldown",
+                 "data": "data",
+                 "encoded_data": "encoded_data",
+                 "predictions": "df",
+                 "analysis": "$runtime_analyzer",
+                 "ts_analysis": "$ts_analysis" if tss.is_timeseries else None,
+                 "target_name": "$target",
+                 "target_dtype": "$dtype_dict[self.target]",
+             },
+         }), ('timeseries_transformer', {
+             "module": "transform_timeseries",
+             "args": {
+                 "timeseries_settings": "$problem_definition.timeseries_settings",
+                 "data": "data",
+                 "dtype_dict": "$dtype_dict",
+                 "target": "$target",
+                 "mode": "$mode",
+             },
+         }), ('timeseries_analyzer', {
+             "module": "timeseries_analyzer",
+             "args": {
+                 "timeseries_settings": "$problem_definition.timeseries_settings",
+                 "data": "data",
+                 "dtype_dict": "$dtype_dict",
+                 "target": "$target",
+             },
+         })]
+
+    for field_name, implicit_value in hidden_fields:
+        field = json_ai.__getattribute__(field_name)
+        if field is None:
+            if tss.is_timeseries or field_name not in ('timeseries_analyzer', 'timeseries_transformer'):
+                field = implicit_value
+        else:
+            args = eval(field['module']).__code__.co_varnames
+            for arg in args:
+                if 'args' not in field:
```

Comment:
I'd prefer blocking these up into small sub-functions but the gist of this is clear